src/App.tsx

```diff
@@ -1,39 +1,68 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
+
+type State = {
+  hasClock: boolean,
+  clockName: string,
+};
+type Props = {
+  clockName: string,
+};
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends React.Component<{}, State, Props> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  timerId = 0;
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+    document.addEventListener('contextmenu',
+      (event) => this.handleContextMenuClick(event));
+    document.addEventListener('click', (event) => this.handleClick(event));
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  componentDidUpdate(_prevState: State, prevProps: Props) {
+    /* eslint-disable no-console */
+    console.debug(`Renamed from ${prevProps.clockName} to ${this.state.clockName}`);
+  }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  componentWillUnmount() {
+    document.removeEventListener('contextmenu', this.handleContextMenuClick);
+    document.removeEventListener('click', this.handleClick);
+  }
 
-        {' time is '}
+  handleContextMenuClick = (event: Event) => {
```

Comment:
Change the event type to `React.MouseEvent` instead of `Event`. Update the line to `handleContextMenuClick = (event: React.MouseEvent) => {`.

Reply:
I have fixed all previous comment which do not on script performance, I can't get rid of 'test' then script does not work, also if I change type for event it stops working
<img width="1280" alt="Screenshot 2023-05-12 at 15 49 20" src="https://github.com/mate-academy/react_clock/assets/20277989/616e7687-f316-4c8a-96d9-134fe7241142">